src/main/java/org/elasticsearch/search/highlight/FastVectorHighlighter.java

```diff
@@ -63,7 +63,7 @@ public HighlightField highlight(HighlighterContext highlighterContext) {
         FetchSubPhase.HitContext hitContext = highlighterContext.hitContext;
         FieldMapper mapper = highlighterContext.mapper;
 
-        if (!(mapper.fieldType().storeTermVectors() && mapper.fieldType().storeTermVectorOffsets() && mapper.fieldType().storeTermVectorPositions())) {
+        if (canHighlight(mapper) == false) {
```

Comment:
I get the comment usually the other way round with the argument that  '== false'  is much easier to read and a '!' is easy to miss so I'd rather leave it this way.